botwfstools/botw_patcher.py

```diff
@@ -156,18 +166,22 @@ def _fix_rstb_resource_size(path: Path, rel_path: Path, table: rstb.ResourceSize
         % (prev_resource_size, resource_size, ' '.join(notes)))
     table.set_size(resource_path, resource_size)
 
-def _get_resource_path_for_rstb(rel_path: Path) -> str:
+def _get_resource_path_for_rstb(rel_path: Path, is_aoc: bool) -> str:
     """Get the RSTB resource path for a resource file."""
     rel_path = rel_path.with_suffix(rel_path.suffix.replace('.s', '.'))
 
     for parent in rel_path.parents:
         if _is_archive_filename(parent):
-            return rel_path.relative_to(parent).as_posix()
+            resource_path = rel_path.relative_to(parent).as_posix()
 
     # File is not in any archive, so just return the path relative to the content root.
-    return rel_path.as_posix()
+    resource_path = rel_path.as_posix()
```

Comment:
Since you've removed the return, the execution will *always* reach this line, meaning the correct value of resource_path for archived files will *always* be overwritten.

I suggest returning keeping the returns and just adding `get_path()` around the returned strings, where `get_path` is a local function that checks for is_aoc and the AoC conditions (see an earlier comment) and adds the prefix if needed. (Feel free to change the name to something better)